--- origsrc/iv-2.2.1/Makefile	2007-07-30 02:42:17.000000000 -0500
+++ src/iv-2.2.1/Makefile	2007-08-03 20:23:02.078125000 -0500
@@ -24,3 +24,5 @@
 	@for subdir in $(ALL_SRC_DIRS); do			\
             $(MAKE) -s -C $$subdir -f Makefile $@; 		\
         done
+
+.PHONY: install
--- origsrc/iv-2.2.1/iv/framesdlg.c	2007-07-30 02:42:17.000000000 -0500
+++ src/iv-2.2.1/iv/framesdlg.c	2007-08-03 20:23:02.093750000 -0500
@@ -1,5 +1,6 @@
 #include <stdlib.h>
 #include <string.h>
+#include <sys/syslimits.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
--- origsrc/iv-2.2.1/iv/platforms.ini	2007-07-30 02:42:17.000000000 -0500
+++ src/iv-2.2.1/iv/platforms.ini	2007-08-03 20:23:02.093750000 -0500
@@ -801,3 +801,123 @@
 			DependType = Program
 			MustExist = Yes
 			DependPath = endeavour2-config
+
+# Cygwin
+#
+Platform = Cygwin
+	Description = For Cygwin 1.5
+	PREFIX = /usr/
+	CFLAGS = -Wall -O6 -funroll-loops -fomit-frame-pointer -ffast-math \
+-finline-functions -DNEED_STRCASESTR
+	LIBS =
+	LIB_DIR =
+	CC = gcc
+	CPP = g++
+	PlatformSearchPathInclude = /usr/include/
+	PlatformSearchPathInclude = /usr/X11R6/include/
+	PlatformSearchPathLib = /usr/lib/
+	PlatformSearchPathLib = /usr/X11R6/lib/
+	PlatformSearchPathEtc = /etc/
+	PlatformSearchPathBin = /usr/bin/
+	PlatformSearchPathBin = /usr/X11R6/bin/
+	PlatformSearchPathData = /usr/share/icons/
+	PlatformFeature = debug
+		Description = Just adds -g to the CFLAGS for debugging
+		MustExist = No
+		FeatureCFLAGS = -g
+	PlatformFeature = title_image
+		Description = Compiles in the title image (increases\
+ size of binary)
+		MustExist = Preffered
+		FeatureCFLAGS = -DINCLUDE_IV_TITLE_IMAGE
+	PlatformFeature = libgif
+		Description = Uncompressed GIF Library
+		MustExist = Preffered
+		FeatureCFLAGS = -DHAVE_LIBGIF
+		FeatureLIBS = -lgif
+		FeatureDepend = libgif-devel
+			DependType = Header
+			MustExist = Yes
+			DependPath = gif_lib.h
+			DependGrepString = DGifOpenFileName
+	PlatformFeature = libjpeg
+		Description = JPEG Library
+		MustExist = Preffered
+		FeatureCFLAGS = -DHAVE_LIBJPEG
+		FeatureLIBS = -ljpeg
+		FeatureDepend = libjpeg-devel
+			DependType = Header
+			MustExist = Yes
+			DependPath = jpeglib.h
+			DependGrepString = jpeg_create_decompress
+	PlatformFeature = libpng
+		Description = PNG Library
+		MustExist = Preffered
+		FeatureCFLAGS = -DHAVE_LIBPNG `libpng-config --cflags`
+		FeatureLIBS = `libpng-config --libs`
+		FeatureDepend = libpng-devel
+			DependType = Header
+			MustExist = Yes
+			DependPath = png.h
+			DependGrepString = png_create_read_struct
+	PlatformFeature = libxpm
+		Description = XPM Library
+		MustExist = Preffered
+		FeatureCFLAGS = -DHAVE_LIBXPM
+		FeatureLIBS = -lXpm
+		FeatureLIB_DIR = -L/usr/X11R6/lib/
+		FeatureDepend = libxpm-devel
+			DependType = Header
+			MustExist = Yes
+			DependPath = X11/xpm.h
+			DependGrepString = XpmReadFileToXpmImage
+	PlatformFeature = Imlib
+		Description = Image Library
+		MustExist = Preffered
+		FeatureCFLAGS = `imlib-config --cflags` -DHAVE_IMLIB
+		FeatureLIBS = `imlib-config --libs`
+		FeatureDepend = Imlib-devel
+			DependType = Program
+			MustExist = Yes
+			DependPath = imlib-config
+	PlatformFeature = Imlib-transpixel-fix
+		Description = Bugfix for Imlib transparent pixel loading
+		MustExist = Preffered
+		FeatureCFLAGS = -DUSE_IMLIB_TRANSPIXEL_FIX
+	PlatformFeature = libXp
+		Description = X Print printing (instead of using lpr for printing)
+		MustExist = Preffered
+		FeatureCFLAGS = -DHAVE_XPRINT
+		FeatureLIBS = -lXp
+		FeatureDepend = libXp-lib
+			DependType = Library
+			MustExist = Yes
+			DependPath = libXp.dll.a
+			DependGrepString = XpQueryExtension
+		FeatureLIB_DIR = -L/usr/X11R6/lib/
+		FeatureDepend = libXp-devel
+			DependType = Header
+			MustExist = Yes
+			DependPath = X11/extensions/Print.h
+			DependGrepString = XpQueryExtension
+	PlatformFeature = libgtk
+		Description = GTK+ 1.2.x
+		MustExist = Yes
+		FeatureCFLAGS = `gtk-config --cflags`
+		FeatureLIBS = `gtk-config --libs`
+		FeatureLIB_DIR = `gtk-config --libs`
+		FeatureDepend = gtk-devel
+			DependType = Program
+			MustExist = Yes
+			DependPath = gtk-config
+	PlatformFeature = libendeavour2
+		Description = Endeavour 2 API
+		MustExist = Preffered
+		FeatureCFLAGS = -DHAVE_LIBENDEAVOUR2 `endeavour2-config --cflags`
+		FeatureLIBS = `endeavour2-config --libs`
+		FeatureLIB_DIR = `endeavour2-config --libs`
+		FeatureDepend = endeavour2-devel
+			DependType = Program
+			MustExist = Yes
+			DependPath = endeavour2-config
+
